components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1812,125 +1816,341 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
+            int indexOfDomainSeparator = attributeValue.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
+
+            // Index less than 1 would indicate that there is not domain appended in front if the attribute value.
+            if (indexOfDomainSeparator > 0) {
+                if (log.isDebugEnabled())
+                    log.debug(String.format("Attribute value: %s is embedded with a domain.", attributeValue));
+                extractedDomain = attributeValue.substring(0, indexOfDomainSeparator).toUpperCase();
+
+                // Check whether the domain name is equal to the extracted domain name from attribute value.
+                if (StringUtils.isNotEmpty(domainName) && StringUtils.isNotEmpty(extractedDomain) && !extractedDomain
+                        .equalsIgnoreCase(domainName)) {
+                    throw new BadRequestException(String.format(
+                            " Domain name: %s in the domain parameter does not match with the domain name: %s in "
+                                    + "search attribute value of %s claim.", domainName, extractedDomain,
+                            attributeName));
+                }
+                // Remove the domain name from the attribute value and update it in the expression node.
+                node.setValue(attributeValue.substring(indexOfDomainSeparator + 1));
+                return extractedDomain;
+            } else {
+                return domainName;
+            }
+        } else {
+            // If the domain is not embedded, return domain name passed in the request.
+            return domainName;
+        }
+    }
+
+    /**
+     * Method to append primary domain name for the roles in the primary user store.
+     *
+     * @param roleName Role name
+     * @return Updated role name for roles in primary domain. Return same value for roles in other domains.
+     */
+    private String appendPrimaryDomainNameForRoleName(String roleName) {
+
+        // Construct the group name with domain -if not already provided, in order to support multiple user
+        // store feature with SCIM.
+        if (roleName.indexOf(CarbonConstants.DOMAIN_SEPARATOR) > 0) {
+            return roleName;
+        } else {
+            // Since the role of primary domain does not contain domain name in front of the role with a domain
+            // separator, Append primary domain name in front of the role for roles in the primary domain.
+            return UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME + CarbonConstants.DOMAIN_SEPARATOR + roleName;
+        }
+    }
+
+    /**
+     * Method to get the role name with attributes.
+     *
+     * @param roleName       Role name
+     * @param filteredGroups List of filtered groups to store the results
+     * @throws CharonException
+     * @throws UserStoreException
+     */
+    private void getRoleWithAttributes(String roleName, List<Object> filteredGroups)
```

Comment:
How are we providing attributes to this method?

Reply:
We cannot provide any required attributes. That part is not implemented. This method returns roles with default set of attributes.

Reply:
Then shall we reflect that `default set of attributes` in the method name?

Reply:
Fixed: 497cb3c10e705b08f73a61c5004190d023109795